### ISSUE TITLE
Docs: Unclear about using oracle-enhanced in multithreading environment (Puma ...)

### ISSUE THREAD

**Issue by ekr1** [rsim/oracle-enhanced#1377]
### Steps to reproduce

Read documentation, search github repository including old issues for the word "thread" or "multithreading".

### Expected behavior

Some notes about using OracleEnhanced in the context of a multithreading webserver like Puma should appear. If it just works out of the box with standard Rails settings (i.e., none specifically, just `rails s puma`), then it would be great to read a short notice about it, here on the frontpage.

It would be especially helpful to know whether OracleEnhanced will give up the GIL right before waiting for low level Oracle I/O, i.e. whether other ruby threads can expect to run while OracleEnhanced is waiting on Oracle responses, and so on. This would make it possible to decide whether to run a OracleEnhanced based application in a multithreaded webserver environment at all.

Also, it would be great to read about whether OracleEnhanced itself (when instantiating and using it directly, without one of the big frameworks like Rails) is thread-safe. I would not assume it is, but it would be nice to have it spelled out clearly to keep the unwary from harm. I *would* assume that it is perfectly save to create and use any amounts of OracleEnhanced instances in a multithreaded environment (only accessing a specific instance from one thread at a time, obviously).

### Actual behavior

Searching the repository for the word "thread" turns up nothing at all.

**Comment by yahonda** [comment]
Hi, When I develop Oracle enhanced adapter code I do not consider any differences if a web server is running as multi-threaded or multi-processed, then there is nothing I can provide right now.
